crates/rethnet_evm/src/mempool.rs

```diff
@@ -224,7 +233,27 @@ impl MemPool {
 
             *old_transactions = transaction;
         } else {
+            let caller = *transaction.caller();
+            let mut next_pending_nonce = transaction.nonce() + 1;
+
             pending_transactions.push(transaction);
+
+            // Move as many future transactions as possible to the pending status
+            if let Some(future_transactions) = self.future_transactions.get_mut(&caller) {
+                while let Some((idx, _)) = future_transactions
```

Comment:
It looks like this might be quadratic in the length of `future_transactions`. I'm wondering if we could maybe partition in one go?

Reply:
Sure, shouldn't be a problem as long as the number of transactions in the mempool is <100 which is I think a safe assumption for now.

Reply:
https://github.com/NomicFoundation/rethnet/issues/181